### DISCUSSION TITLE
Ring matches in rdSubstructLibrary.GetMatches

### DISCUSSION THREAD

**Discussion by PaulaKramer** [rdkit/rdkit#7298] [Q&A]
Hi all, 
I have implemented a substructure search using the `rdSubstructLibrary`. Is there a way to prevent non-ring queries from matching with rings? I included a short code example where the substructure matches, but I would like to exclude cases like these. 
I think I’m looking for something similar to the `rdFMCS.FindMCS` arguments `ringMatchesRingOnly` or `completeRingsOnly`. 

```
from rdkit import Chem
from rdkit.Chem import rdSubstructLibrary

# create example library
mol = Chem.MolFromSmiles("CC1CC1")
library = rdSubstructLibrary.SubstructLibrary()
library.AddMol(mol)

# example query 
query = Chem.MolFromSmiles("CCCC")
indices = library.GetMatches(Chem.AdjustQueryProperties(query))
print(len(indices))
```

Thanks for your help :) 
Paula

**Comment by bp-kelley** [comment]
It looks like what you want is adjustRingCount and adjustRingChain

```
params = Chem.AdjustQueryParameters()
params.adjustRingCount = True
params.adjustRingChain = True
indices = library.GetMatches(Chem.AdjustQueryProperties(query, params))
print(len(indices))
```

```
0
```

This will generate the query smarts:

```
q = Chem.AdjustQueryProperties(query, params)
Chem.MolToSmarts(q)
```

```
'[C&R0&!R][C&R0&!R][C&R0&!R][C&R0&!R]'
```

Which specifies the atoms not starting in a ring will not match non ring atoms.  This is not the same as complete rings in the rdFMCS but perhaps close enough.
